File: exercism/kotlin/medium/binary-search/src/main/kotlin/BinarySearch.kt
object BinarySearch {
    fun search(list: List<Int>, target: Int) : Int {
        var left = 0
        var right = list.lastIndex
        while (left <= right) {
            var middle_index = (left + right) / 2
            var middle_value = list[middle_index]
            when {
                middle_value == target -> return middle_index 
                middle_value < target -> left = middle_index + 1 
                else -> right = middle_index - 1
            }
        }
        return -1
    }
}

File: exercism/csharp/easy/binary-search/BinarySearch.cs
using System;

public static class BinarySearch
{
    public static int Find(int[] input, int value)
    {
        var left = 0;
        var right = input.Length - 1;
        while (left <= right)
        {
            var middle = (left + right) / 2;
            if (input[middle] < value)
            {
                left = middle + 1;
                continue;
            }
            if (input[middle] > value)
            {
                right = middle - 1;
                continue;
            }
            return middle;
        }
        return -1;
    }
}